padertorch/train/hooks.py

```diff
@@ -1022,3 +1023,46 @@ def set_value(self, trainer, value):
         optimizer = self.get_optimizer(trainer)
         for param_group in optimizer.param_groups:
             param_group['lr'] = value
+
+
+class EmissionsTrackerHook(TriggeredHook):
+    @property
+    def priority(self):
+        return Priority.SUMMARY
+
+    def __init__(self, trigger, prefix='x_emissions', storage_dir=None):
+        super().__init__(trigger)
+        self.prefix = prefix
+        self.storage_dir = storage_dir
+        from codecarbon import EmissionsTracker
+        self.tracker = EmissionsTracker(
+            output_dir=storage_dir, on_csv_write="update", log_level='error')
+        self.tracker.start()
+
+    def pre_step(self, trainer: 'pt.Trainer'):
+        if self.trigger(iteration=trainer.iteration, epoch=trainer.epoch):
+            self.tracker.flush()
+            self.dump_emissions(trainer)
+
+    def dump_emissions(self, trainer):
+        emissions_file = Path(self.storage_dir) / 'emissions.csv'
+        with emissions_file.open() as fid:
+            lines = fid.read().split('\n')
+        if len(lines) < 3:
+            return
+        lines = lines[:-1]
```

Comment:
How often does the tracker update the file? Does this catch all updates?

Reply:
it is updated when the flush or stop method is called, which the hook does right before calling dump_emissions